src/main/resources/io/github/dfdlschemas/envelopepayload/xsd/envelopePayload.dfdl.xsd

```diff
@@ -0,0 +1,94 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<schema
+  xmlns="http://www.w3.org/2001/XMLSchema"
+  xmlns:xs="http://www.w3.org/2001/XMLSchema"
+  xmlns:fn="http://www.w3.org/2005/xpath-functions"
+  xmlns:dfdl="http://www.ogf.org/dfdl/dfdl-1.0/"
+  xmlns:ep="urn:io.github.dfdlschemas.envelopepayload"
+  targetNamespace="urn:io.github.dfdlschemas.envelopepayload"
+  xmlns:msHdr="urn:milstd2045DFDL"
+  xmlns:pcap="urn:pcap:2.4"
+  elementFormDefault="unqualified">
+
+  <import namespace="urn:milstd2045DFDL"
+          schemaLocation="/com/owlcyberdefense/mil-std-2045/xsd/milstd2045_application_header.dfdl.xsd"/>
+  <import namespace="urn:pcap:2.4"
+          schemaLocation="/com/tresys/pcap/xsd/pcap.dfdl.xsd"/>
+
+  <annotation>
+    <appinfo source="http://www.ogf.org/dfdl/">
+      <dfdl:format ref="msHdr:msCommon"/>
+      <dfdl:defineVariable name="filename" type="xs:string"/>
+      <dfdl:defineVariable name="binaryFileKind" type="xs:string"/>
+    </appinfo>
+  </annotation>
+
+  <group name="payloadContentGroup">
+    <sequence>
+      <element name="milstd2045_application_header" type="msHdr:milstd2045_application_header_type"/>
+      <element name="payload">
+        <complexType>
+          <choice dfdl:choiceDispatchKey='{
+               ../milstd2045_application_header/contents/message_handling_group/item[1]/umf/value
+            }'>
+            <element dfdl:choiceBranchKey="OK_Binary_File"
+                     name="binary_file" type="ep:binary_file_type"/>
+            <!--
+            This format only supports the mil-std-2045 header with binary_file payload type.
+            Any other value for the umf/value in the header will cause parsing to fail here
+            because there are no other choice branches.
+            -->
+          </choice>
+        </complexType>
+      </element>
+      <sequence dfdl:alignment="1" dfdl:alignmentUnits="bytes"/><!-- force realignment -->
+    </sequence>
+  </group>
+
+  <complexType name="binary_file_type">
+    <sequence>
+      <sequence>
```

Comment:
Is there a practical reason for having two sequences with two annotations, instead of one sequence with one annotation containing one dfdl:assert and two dfdl:newVariableInstances?

Reply:
Yes actually. 

I hate this thing about DFDL, but the annotation block at the start of a sequence, any asserts/discriminators in it evaluate AFTER the sequence has been parsed. That's so that expressions in them can refer to the contents of the sequence itself, because they'll exist after parsing.  (Section 9.5 of DFDL spec)

But executing AFTER the sequence, when they're written lexically at the top of the sequence content, is just so misleading..... I believe this was a big mistake. It's also only dfdl:assert/discriminator with testKind 'expression' that execute after the sequence contents. Assert/discriminator with testKind 'pattern' execute before the sequence content, as well as setVariable, newVariableInstance, etc. This was a language design mistake plain and simple. I think the purpose was symmetry with assert/discriminator on elements, which execute after the element value has been determined. But the two cases really don't need to be consistent. 

So anyway, I use a style that says you always put statement annotations inside their own otherwise empty sequence. Then you place that sequence in the enclosing sequence at the location where you want it executed. At the top for execution before any of the sequence content, last if you want it after all the sequence content. 

The other interaction is that newVariableInstance creates a variable instance only until the sequence containing it ends. So it has to be in a separate sequence from the assert above so that it can surround the scope of content that contains that variable's usage. 

If we had just made the rule "everything evaluates before the sequence content" then all these annotations could just have been on the one enclosing sequence, and all this nesting of inner sequences would not be needed. 

Oh well. 

I will add a comment to the inner sequence like 
```
<!-- execute this assert before the sequence contents --> 
```